include/nbl/video/ILogicalDevice.h

```diff
@@ -362,6 +362,32 @@ class ILogicalDevice : public core::IReferenceCounted
         return createGPUDescriptorSet_impl(pool, std::move(layout));
     }
 
+    core::smart_refctd_ptr<IDescriptorPool> createDescriptorPoolForDSLayouts(core::smart_refctd_ptr<IGPUDescriptorSetLayout>* begin, core::smart_refctd_ptr<IGPUDescriptorSetLayout>* end)
+    {
+        size_t setCount = end - begin;
+        std::vector<IDescriptorPool::SDescriptorPoolSize> poolSizes;
```

Comment:
precompute number of unique descriptor types and use core::smart_refctd_dynamic_array instead